lib/job/arBaseJob.class.php

```diff
@@ -263,4 +292,31 @@ protected function signOut()
     QubitAcl::destruct();
     $this->user->signOut();
   }
+
+  /**
+   * Check if another sensitive job is running.
+   *
+   * @return boolean true if this is the oldest being executed
+   *                 false if there is an older one in execution
+   */
+  protected function canBeFullyExecuted()
+  {
+    // Add job names directly to the query to avoid params escaping
+    $jobNames = "('" . implode("','", $this->avoidParallelExecutionJobs) . "')";
+
+    // Select sensitive jobs running ordering by created_at
+    $sql = "SELECT job.id FROM job
+     LEFT JOIN object ON object.id = job.id
+     WHERE job.status_id = :statusId
+     AND job.name IN $jobNames
+     ORDER BY object.created_at;";
+
+    $params = array(':statusId' => QubitTerm::JOB_STATUS_IN_PROGRESS_ID);
+
+    // It will be at least one sensitive job running, this one
+    $runningJobs = QubitPdo::fetchAll($sql, $params, array('fetchMode' => PDO::FETCH_ASSOC));
+
+    // If this job is the first one, it can be fully executed
+    return $this->job->id === $runningJobs[0]['id'];
```

Comment:
Good catch Steve, the JobsDeleteAction doesn't delete running jobs, but I didn't know about this task and it looks like it's only showing a warning, so running jobs could be actually deleted, I'd add check. But in that case something else bad is going to happen ...